packages/react-components/src/components/SendBox.tsx

```diff
@@ -217,12 +242,36 @@ export const SendBox = (props: SendBoxProps): JSX.Element => {
   };
 
   /* @conditional-compile-remove(file-sharing) */
-  const renderFileUploadErrorMessage: JSX.Element = useMemo(() => {
+  const uploadInProgressErrorBar = useMemo(() => {
```

Comment:
SendBox logic is getting complicated now - I am worried that it could become next messageThread - which is hard to read and maintain since code pieces are flying all around in the same file. And there is a lot /* @conditional-compile-remove(file-sharing) */ which is hard to follow the stable version logic now

I believe we could separate this logic and all hooks, put them into one single component maybe name it FileSharingSendBoxErrorComponent or so, then the hierarchy is:
SendBox -> FileSharingSendBoxErrorComponent (choose which fileUploadError to be rendered) -> SendBoxErrorBar

The idea is keeping each part small and easy to read and easy for refactoring

Reply:
Will do